sdk/python/feast/client.py

```diff
@@ -568,36 +590,186 @@ def ingest(
         else:
             raise Exception(f"Feature set name must be provided")
 
+        time_start = time.time()
+
+        # Read table and get row count
         table = _read_table_from_source(source)
 
-        # Update the feature set based on DataFrame schema
-        if force_update:
-            # Use a small as reference DataFrame to infer fields
-            ref_df = table.to_batches(max_chunksize=20)[0].to_pandas()
+        # Delete source from memory after conversion into PyArrow table
+        del source
```

Comment:
In the latest commit, the user provided source (DataFrame/Json/csv/ParquetFile) will be transformed into another Parquet file with a row_group_size equal to the maximum chunk size to be passed to each pool worker.

The index for these chunks within the Parquet file are passed to the pool workers.

As such, the main source is no longer being referenced and used. The Parquet file that has row_groups will be used instead.

Reply:
Ok, meaning the provided dataframe from the user is not deleted?

Reply:
Alright, i see what you mean. It will be deleted unfortunately. 

I'll remove that line to make sure that it will not be removed. As such, we will be fully reliant on GC to clear the memory.

Reply:
fixed